cranelift/codegen/src/isa/aarch64/inst/args.rs

```diff
@@ -220,8 +220,11 @@ impl AMode {
             &AMode::RegExtended(r1, r2, ext) => {
                 AMode::RegExtended(allocs.next(r1), allocs.next(r2), ext)
             }
-            &AMode::PreIndexed(reg, simm9) => AMode::PreIndexed(allocs.next_writable(reg), simm9),
-            &AMode::PostIndexed(reg, simm9) => AMode::PostIndexed(allocs.next_writable(reg), simm9),
+            // Note that we assert that `reg` is SP, so we don't
+            // provide it to the regalloc and don't get an alloc in
+            // return.
+            &AMode::PreIndexed(reg, simm9) => AMode::PreIndexed(reg, simm9),
```

Comment:
I don't really like this change because these are valid addressing modes with arbitrary registers, which admittedly we don't use at the moment. IMHO a cleaner approach is probably to introduce `SPPreIndexed` and `SPPostIndexed` variants that only have an immediate parameter (hopefully on top of PR #4832 :smile:) - we can also remove the current ones, to be on the safe side.

Reply:
Renamed to `SPPreIndexed` / `SPPostIndexed`. Happy to eventually make it more general again; my concern here was mainly to avoid the need to include input/output registers for now, for simplicity.